js/app/dashboard/store/reducers.js

```diff
@@ -0,0 +1,116 @@
+import { combineReducers } from 'redux'
+import _ from 'lodash'
+
+
+// initial data pumped from the template
+const tasksInitial = window.AppData.Dashboard.tasks,
+      unassignedTasksInitial = _.filter(tasksInitial, task => !task.isAssigned),
+      assignedTasksList = _.filter(tasksInitial, task => task.isAssigned),
+      assignedTasksByUserInitial = _.groupBy(assignedTasksList, task => task.assignedTo)
+
+/*
+  Store for all assigned tasks
+*/
+const assignedTasksByUser = (state = assignedTasksByUserInitial, action) => {
+
+  let  userTasks = state[action.username],
+    newState = { ...state },
+    newUserTasks,
+    taskComparator = (taskA, taskB) => taskA['@id'] === taskB['@id']
+
+  switch(action.type) {
+    case 'ASSIGN_TASKS':
+      newUserTasks = userTasks.slice()
+      newState[action.username] = Array.prototype.concat(newUserTasks, action.tasks)
+      newState[action.username].duration = 0
+      newState[action.username].distance = 0
+      break
+    case 'REMOVE_TASKS':
+      newUserTasks  = _.differenceWith(
+        userTasks,
+        _.intersectionWith(userTasks, action.tasks, taskComparator),
+        taskComparator
+      )
+      newState[action.username] = newUserTasks
+      newState[action.username].duration = 0
+      newState[action.username].distance = 0
+      break
+    case 'ADD_USERNAME':
+      newState[action.username] = []
+      newState[action.username].duration = 0
+      newState[action.username].distance = 0
+      break
+    case 'SAVE_USER_TASKS_SUCCESS':
+      newState[action.username].duration = action.duration
+      newState[action.username].distance = action.distance
+      break
+    case 'UPDATE_TASK':
+      if (action.task.assignedTo) {
+        userTasks = state[action.task.assignedTo]
+        let index =  _.findIndex(userTasks, action.task, taskComparator)
+        newState[action.task.assignedTo] = Array.prototype.splice(index, 1, action.task)
```

Comment:
When the 2nd parameter is greater than 0, `Array.prototype.splice` _removes_ elements. Also, `Array.prototype.splice` returns the number of deleted elements, not the array. 

I think it should be:

```
userTasks.splice(index, 0, action.task)
newState[action.task.assignedTo] = userTasks
```